### DISCUSSION TITLE
Discussion of the content of issue and pull request templates

### DISCUSSION THREAD

**Discussion by jh-RLI** [openego/oeplatform#1159] [General]
Soon I will implement [templates](https://github.blog/2016-02-17-issue-and-pull-request-templates/) for both issues and pull requests. I came across this page where the author proposes the structure for these templates. I would like to discuss them with you and get feedback if you think they are sutable or are too much / not sutable / something missing?

I think they cover our use cases. see both templates below:

# Issues (features and bugs - maybe better to offer seperate templates for bugs and feature issues?)
source: https://www.talater.com/open-source-templates/#/page/98

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Version used:
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

# Pull Request
source: https://www.talater.com/open-source-templates/#/page/99

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

**Comment by jh-RLI** [comment]
Assignments are missing:
- Who will work on the task
- Who can review the result 
- Who can guide with technical questions 
- Who can performe a release

**Comment by jh-RLI** [comment]
Additionaly there should be some task automation added
- When creating an issue automatiaccly add it to a triage Project (similar to what is implemented in the OEO)
https://github.com/OpenEnergyPlatform/ontology/projects/3

**Comment by Ludee** [comment]
In general I like the templates and we already implemented them for the OEO.

https://github.com/OpenEnergyPlatform/ontology/tree/dev/.github/ISSUE_TEMPLATE
https://github.com/OpenEnergyPlatform/ontology/blob/dev/.github/pull_request_template.md
https://github.com/OpenEnergyPlatform/ontology/pull/1162

The given example is quite large, in practice I prefer to concentrate on the key points.
Do not use more than 3 checkboxes or they won't be used seriously anyway.

**Reply by Ludee** [reply]
I just pushed the merged templates to this repo:
https://github.com/rl-institut/super-repo/pull/15

**Reply by jh-RLI** [reply]
Nice will have a look at it. I also think the OEO templates are great. Maybe we just need to refine them a bit to focus more on software than ontology development.

**Reply by jh-RLI** [reply]
Ah i see you already refined them :)
